File: app/src/main/java/ladsoft/com/popularmoviesapp/core/mvp/Mvp.java
package ladsoft.com.popularmoviesapp.core.mvp;

public interface Mvp {

        interface Presenter<V> {
                void attachView(V view);

                void detachView();

                boolean isViewAttached();

                V getView();
        }

        interface Model<T extends Model.ModelCallback> {
                void attach(T callback);
                void detach();
                boolean isAttached();

                interface ModelCallback {}
        }
}

File: app/src/main/java/ladsoft/com/popularmoviesapp/core/mvp/Mvp.kt
package ladsoft.com.popularmoviesapp.core.mvp

interface Mvp {

    interface Presenter<V> {
        fun attachView(view: V)

        fun detachView()

        val isViewAttached: Boolean

        fun getView(): V?
    }

    interface Model<in T : Model.ModelCallback> {
        fun attach(callback: T)
        fun detach()
        val isAttached: Boolean

        interface ModelCallback
    }
}
